app/src/main/java/org/sagebase/crf/step/active/HeartRateCamera2Recorder.java

```diff
@@ -250,18 +255,23 @@ public void setRecorderListener(RecorderListener listener) {
     public void stop() {
         subscriptions.unsubscribe();
         heartBeatJsonWriter.stop();
+
         if (mediaRecorderFile.exists()) {
             FileResult fileResult = new FileResult(fileResultIdentifier(), mediaRecorderFile, MP4_CONTENT_TYPE);
             fileResult.setStartDate(startTime.toDate());
             fileResult.setEndDate(new Date());
-            getRecorderListener().onComplete(this, fileResult);
+            // joliu test
```

Comment:
Will do additional testing on the stop/cancel changes

Reply:
If you remove this line, how does the heart rate step layout know when it is safe to proceed to the next step?

Reply:
The JsonArrayDataRecorder also calls onComplete once its DataLogger finishes writing to disk. Turns out we should call onComplete for the camera, and the step layout should proceed when the json finishes writing.